report.md

```diff
@@ -64,6 +61,104 @@ Plan for refactoring complex code:
+
+Method EqualsBuilder::append(Object, Object):
+
+- CCN: 17
+
+- Problem: it's not too easy to refactor this one. One way is to try to make sense of the type dispatching. But this may caus tests on this method to fail (it did in my attempt at refactoring it). Another is to pick the first lines returning `this` if the arrays are null, known to be different already, or their pointers are the same for a new method
+  
+  ```java
+  // Returns true if the objects are known to be non-equal, if any is null, or if they are
+  // the same pointer
+  
+  private boolean checkNoWorkToBeDone(Object lhs, Object rhs) {
+      if (!isEquals) {
+          return true;
+      }
+      if (lhs == rhs) {
+          return true;
+      }
+      if (lhs == null || rhs == null) {
+          this.setEquals(false);
+          return true;
+      }
+      return false;
+  }
+  ```
+  
+  This stilll causes an error, but because one test checks how long it takes to do an operation and with the new method it takes longer. We will not implement this one.
+
+Method ModuleHandler::adjustModuleGraph(Class, Class, boolean, boolean):
+
+- CCN: 13
+
+- Possible refactoring: a few parts can be extracted into separate functions, e.g.
+  
+  - ```java
+    while (!targetVisible && classLoader != null) {
+                    classLoader = classLoader.getParent();
+                    targetVisible = classLoader == target.getClassLoader();
+    }
+    ```
+
+```java
+if (needsExport) {
+                implementation =
+                        implementation.andThen(
+                                MethodCall.invoke(addExports)
+                                        .onMethodCall(sourceLookup)
+                                        .with(target.getPackage().getName())
+                                        .withMethodCall(targetLookup));
+            }
+            if (needsRead) {
+                implementation =
+                        implementation.andThen(
+                                MethodCall.invoke(addReads)
+                                        .onMethodCall(sourceLookup)
+                                        .withMethodCall(targetLookup));
+            }
+```
+
+    This should help to decrease the CCN while not changing the behaviour.
+
+Method SubclassBytecodeGenerator::mockClass(MockFeatures):
+
+- CCN: 34
+
+- Possible refactoring: extract the following code lines into dedicated methods (it should reduce the CCN, but increase the time for execution by a bit):
+  
+  - 139-152: this is a handler in case we don't have the same class loader yet, return a boolean and assign it to `shouldIncludeContextLoader`, change the break for a return.
+  
+  - 166-223: extract this into a `handleMockLocalOrNot(MockFeatures<T> features, ClassLoader classLoader)` method and give `features` as argument (should be of void return type)
```

Comment:
'handleMockLocalOrNot should return a boolean to assign to localMock?

Reply:
No, as we can extract the whole part dealing with the fact it is a local mock (ending at line 223). It should return a String because we need the name created in this part. Or we extract in several functions.